File: 0645-set-mismatch/solution.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int i=0;
        while(i<nums.length){
                int correct = nums[i] - 1;
            if(nums[i] != nums[correct]){
                swap(nums, i, correct);
            } 
             else{
                i++;
            }
        }

        for(int in=0; in<nums.length; in++){
            if(nums[in] != in+1){
                return new int[] {nums[in], in+1};
            }
        }
        return new int[] {-1,-1};
    }

    private void swap(int[] nums, int first, int second){
        int temp = nums[first];
        nums[first] = nums[second];
        nums[second] = temp;
    }
}

File: 0645-set-mismatch/solution.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        d = -1
        m = -1

        for n in nums:
            val = abs(n)
            if nums[val-1] < 0:
                d = val
            else:
                nums[val - 1] *= -1
        
        for i in range(len(nums)):
            if nums[i] > 0:
                m = i+1
        
        return [d,m]

        
